veniq/dataset_collection/validation.py

```diff
@@ -0,0 +1,284 @@
+        filepath: str) -> Tuple[int, int]:
+    """
+    Finds start and end lines for opportunity
+
+    :param filepath: filename where opportunity was found
+    :param extracted_lines_of_opportunity: list of lines for opportunity
+    :return: list of extracted lines for opportunity
+    """
+    start_line_opportunity = min(extracted_lines_of_opportunity)
+    end_line_opportunity = max(extracted_lines_of_opportunity)
+    text = read_text_with_autodetected_encoding(filepath).split('\n')
+    extraction = text[start_line_opportunity:end_line_opportunity]
+    open_brackets = 0
+    close_brackets = 0
+    for x in extraction:
+        close_brackets += x.count('}')
+        open_brackets += x.count('{')
+
+    if open_brackets < close_brackets:
+        diff = close_brackets - open_brackets
+        count = 1
+        for text_line in text[end_line_opportunity:]:
+            if diff < 1:
+                break
+            else:
+                if text_line.find('{') > -1:
+                    diff -= 1
+                    count += 1
+
+        start_line_opportunity += count - 1
+
+    elif open_brackets > close_brackets:
+        diff = open_brackets - close_brackets
+        count = 1
+        for text_line in text[end_line_opportunity:]:
+            if diff < 1:
+                break
+            else:
+                if text_line.find('}') > -1:
+                    diff -= 1
+                    count += 1
+
+        end_line_opportunity += count - 1
+
+    return start_line_opportunity, end_line_opportunity
+
+
+# flake8: noqa: C901
+def validate_row(dataset_dir: Path, row: pd.Series) \
+        -> List[RowResult]:
+    """
+    Validate row of dataset
+
+    :param dataset_dir: directory to dataset, path before the relative path in
+    output_filename
+    :param row: row of dataframe
+    :return: boolean value whether we should consider this row or skip it,
+    Stats - return collected stats
+    """
+    results = []
+    try:
+        start_line_of_inserted_block = int(row[1]['inline_insertion_line_start'])
+        end_line_of_inserted_block = int(row[1]['inline_insertion_line_end'])
+
+        src_filename = row[1]['output_filename']
+        class_name = row[1]['class_name']
+        full_path = dataset_dir / src_filename
+        ast = AST.build_from_javalang(build_ast(full_path))
+        function_to_analyze = row[1]['method_where_invocation_occurred']
+
+        for class_decl in ast.get_proxy_nodes(ASTNodeType.CLASS_DECLARATION):
+            if class_decl.name != class_name:
+                continue
+            elif class_decl.name == class_name:
+                objects_to_consider = list(class_decl.methods) + list(class_decl.constructors) or []
+                for ast_node in objects_to_consider:
+                    result = RowResult(
+                        output_filename=full_path,
+                        input_filename=row[1]['input_filename'],
+                        class_name='',
```

Comment:
why class_name and method_name empty?

Reply:
it's a default initialization. It will be still empty if it fails by some reason. But we can add a result and understand that it was failed. (e.g., Java syntax error)

Reply:
i dont get, when in this part of the code ``except Exception as e:'' u don't add class and method names, or do u?

Reply:
discussed- > added it